phalcon/mvc/model.zep

```diff
@@ -4154,24 +4151,66 @@ abstract class Model implements EntityInterface, ModelInterface, ResultInterface
 			return value;
 		}
 
-		/**
-		 * Check if the property has setters
-		 */
-		let method = "set" . camelize(property);
+		// Use possible setter.
+		if this->_possibleSetter(property, value) {
+			return value;
+		}
 
-		if method_exists(this, method) {
-			return this->{method}(value);
+		// Throw an exception if there is an attempt to set a non-public property.
+		if !this->_isVisible(property) {
+			throw new Exception("Property '" . property . "' does not have a setter.");
 		}
 
-		/**
-		 * Fallback assigning the value to the instance
-		 */
 		let this->{property} = value;
 
 		return value;
 	}
 
 	/**
+	 * Check for, and attempt to use, possible setter.
+	 *
+	 * @param string property
+	 * @param mixed value
+	 * @return string
+	 */
+	protected final function _possibleSetter(string property, value)
+	{
+		var possibleSetter;
+
+		let possibleSetter = "set" . camelize(property);
+		if method_exists(this, possibleSetter) {
+			this->{possibleSetter}(value);
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * Check whether a property is declared private or protected.
+	 * This is a stop-gap because we do not want to have to declare all properties.
+	 *
+	 * @param string property
+	 * @return boolean
+	 */
+	protected final function _isVisible(property)
+	{
+        var reflectionClass, reflectionProp, e;
+
+		//Try reflection on the property.
+		let reflectionClass = new \ReflectionClass(this);
+		try {
+			let reflectionProp = reflectionClass->getProperty(property);
+			if !reflectionProp->isPublic() {
+				return false;
+			}
+		} catch \Exception, e {
+			// The property doesn't exist.
+			return true;
```

Comment:
@sergeyklay the reason this is return true, is because it looks like phalcon models depend on runtime declaration if public properties. In the unit-test model Robots, there are no column properties defined.

If this was false, then the tests would fail, because   undeclared properties would not be set, and the robots model would not get fields set on load. Furthermore, if it were to return false, then the whole function would be redundant and we would have the same functionality removing the final assignment call in __set.

By catching the thrown exception and returning true, we're saying, "if there is no property, make one and assume it's public."

Is there a third party to review this pull request to ensure it is the direction phalcon wants to go with this?